core/src/main/scala/kafka/log/Log.scala

```diff
@@ -229,9 +229,11 @@ class Log(val dir: File,
                                      initFileSize = this.initFileSize(),
                                      preallocate = config.preallocate))
     } else {
-      recoverLog()
-      // reset the index size of the currently active log segment to allow more entries
-      activeSegment.index.resize(config.maxIndexSize)
+      if (!dir.getAbsolutePath.endsWith(Log.DeleteDirSuffix)) {
```

Comment:
Can you avoid this condition by filtering out the directory earlier in the `LogManager`?